chia/full_node/full_node.py

```diff
@@ -519,7 +520,8 @@ async def short_sync_batch(self, peer: WSChiaConnection, start_height: uint32, t
             )
             if first is None or not isinstance(first, full_node_protocol.RespondBlock):
                 self.sync_store.batch_syncing.remove(peer.peer_node_id)
-                raise ValueError(f"Error short batch syncing, could not fetch block at height {start_height}")
+                self.log.error(f"Error short batch syncing, could not fetch block at height {start_height}")
```

Comment:
This is related to one of the issues I mentioned in the PR description. When reorging to a chain with a lower height (but obviously higher weight), we end up requesting *our* height, no their height. This is most likely a bug. I have not investigated this, but the response is a `None` response. When we get the none-response here, we throw. Throwing here terminates all syncing and the long reorg test would stall.

This is the minimal change to recover and fall back to the next sync mechanism, sync from 0.

You may notice that this function returns a boolean whether it was successful or not, so returning `False` is a failure that permits fallback rather than terminating.

Reply:
> are you sure this doesn't open up more bugs ?

No, I'm not sure about that. But it doesn't break any tests, so presumably it doesn't break anything important enough to be tested. It seems pretty safe to me though, as it's merely the *kind* of error that's changed.

> sounds like we dont know what the behavior is

The fundamental error is that we make an invalid request to a peer. This change only addresses how we react to the failure response we get.